xbmc/video/guilib/VideoStreamSelectHelper.cpp

```diff
@@ -0,0 +1,259 @@
+/*
+ *  Copyright (C) 2024 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "VideoStreamSelectHelper.h"
+#include "dialogs/GUIDialogSelect.h"
+#include "ServiceBroker.h"
+#include "guilib/GUIComponent.h"
+#include "guilib/GUIWindowManager.h"
+#include "utils/log.h"
+#include "application/ApplicationPlayer.h"
+#include "application/ApplicationComponents.h"
+#include "FileItemList.h"
+#include "FileItem.h"
+#include "utils/LangCodeExpander.h"
+#include "guilib/LocalizeStrings.h"
+
+void KODI::VIDEO::GUILIB::OpenDialogSelectVideoStream()
+{
+  CGUIDialogSelect* dialog{CServiceBroker::GetGUI()->GetWindowManager().GetWindow<CGUIDialogSelect>(
+      WINDOW_DIALOG_SELECT_VIDEO_STREAM)};
+  if (!dialog)
+  {
+    CLog::LogF(LOGERROR, "Unable to get WINDOW_DIALOG_SELECT_VIDEO_STREAM dialog instance");
+    return;
+  }
+
+  auto& components = CServiceBroker::GetAppComponents();
+  auto appPlayer = components.GetComponent<CApplicationPlayer>();
+  const int streamCount = appPlayer->GetVideoStreamCount();
+  const int selectedIndex = appPlayer->GetVideoStream();
+
+  CFileItemList itemsToDisplay;
+
+  // cycle through each stream and add it to our entry list
+  for (int i = 0; i < streamCount; ++i)
+  {
+    VideoStreamInfo info;
+    appPlayer->GetVideoStreamInfo(i, info);
+
+    std::string languageDesc;
+    g_LangCodeExpander.Lookup(info.language, languageDesc);
+
+    CFileItemPtr fileItem = std::make_shared<CFileItem>();
+    fileItem->SetLabel(info.name);
```

Comment:
construction and setting the label can be done in one step:

`const auto fileItem{std::make_shared<CFileItem>(info.name)};`